modules/light-clients/06-solomachine/light_client_module_test.go

```diff
@@ -1352,6 +1350,9 @@ func (suite *SoloMachineTestSuite) TestVerifyClientMessageHeader() {
 				func() {
 					// store in temp before assigning to interface type
 					cs := sm.ClientState()
+
+					suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.chainA.GetContext(), clientID, sm.ClientState())
+
 					oldPubKey := sm.PublicKey
 					h := sm.CreateHeader(sm.Diversifier)
```

Comment:
`sm.CreateHeader` creates a new header and then increments solomachine timestamp. Previously, we then set below the new solomachine client state with updated timestamp causing failure on due to - `header.Timestamp < cs.ConsensusState.Timestamp` in `verifyHeader`